locks/dynamodb/dynamo_lock_table.go

```diff
@@ -129,9 +129,19 @@ func waitForTableToBeActiveWithRandomSleep(tableName string, client *dynamodb.Dy
 func removeItemFromLockTable(itemId string, tableName string, client *dynamodb.DynamoDB) error {
 	// TODO: should we check that the entry has our own metadata and not someone else's?
 
-	_, err := client.DeleteItem(&dynamodb.DeleteItemInput{
-		Key: createKeyFromItemId(itemId),
-		TableName: aws.String(tableName),
+	// Verify the table actually exists
+	_, err := client.DescribeTable(&dynamodb.DescribeTableInput{TableName: aws.String(tableName)})
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// Unless you specify conditions, the DeleteItem is an idempotent operation;
+	// running it multiple times on the same item or attribute does not result in an error response.
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/#DynamoDB.DeleteItem
+	_, err = client.DeleteItem(&dynamodb.DeleteItemInput{
+		Key:                 createKeyFromItemId(itemId),
+		TableName:           aws.String(tableName),
+		ConditionExpression: aws.String(fmt.Sprintf("attribute_exists(%s)", ATTR_STATE_FILE_ID)),
```

Comment:
Is it possible to specify a `ConditionExpression` that checks if the table exist so we don't have to do a separate check in the previous line?